webapp/src/features/adminEditOrganization/adminEditOrganizationSlice.test.js

```diff
@@ -12,3 +31,202 @@ describe('admin add User reducer', () => {
     })
   })
 })
+
+describe('async thunks', () => {
+  const initialState = {
+    loading: null,
+    value: {
+      successMsg: null,
+      errorState: null,
+      errorMsg: null,
+    },
+  }
+
+  beforeAll(() => {
+    jest.mock('../../apis/api-helper')
+  })
+
+  afterAll(() => {
+    jest.unmock('../../apis/api-helper')
+  })
+
+  describe('editOrganization', () => {
+    it('returns and calls the api correctly', async () => {
+      let dispatch = jest.fn()
+      const getState = jest.fn().mockReturnValue({
+        user: {
+          value: {
+            authLoginData: { token: 'token' },
+          },
+        },
+      })
+      const json = { name: 'orgName' }
+      const selectedOrg = 'selectedOrg'
+      let setValue = jest.fn()
+      let updateOrganizationData = jest.fn()
+      const action = editOrganization({ json, selectedOrg, setValue, updateOrganizationData })
+
+      global.setTimeout = jest.fn((cb) => cb())
+      try {
+        apiHelper._patchData = jest.fn().mockReturnValue({ status: 200, message: 'message' })
+        let resp = await action(dispatch, getState, undefined)
+        expect(resp.payload).toEqual({ success: true, message: 'Changes were successfully applied!' })
+        expect(apiHelper._patchData).toHaveBeenCalledWith({
+          url: EnvironmentVars.adminOrg,
+          token: 'token',
+          body: JSON.stringify(createJsonBody(json, selectedOrg)),
+        })
+        expect(global.setTimeout).toHaveBeenCalledTimes(1)
+        expect(dispatch).toHaveBeenCalledTimes(1 + 2)
```

Comment:
Why the addition in here?

Reply:
The async thunks call dispatch a few times on their own, always twice. So anywhere where I am checking the number of calls, I add the number of calls I want (in this case 1) to the additional 2 that will be called by default. This makes it easy to see what I am checking, compared to just putting 3. This happens all over the place, so I am not sure where to document it

Reply:
Added description the webapp README